### PR TITLE
feat: notebook timestamp nodes

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Follow on from https://github.com/PostHog/posthog/pull/16530

Towards https://github.com/PostHog/posthog/issues/15680

We don't currently support any type of comments in recordings

## Changes

- Introduces a new 'Timestamp' node in TipTap.
    - This stores reference to a session recording & playback time
- Adds a "Comment" button to recordings in notebooks that will insert the new Timestamp node
    - Sets the playback time to the current player time
- Adds a generic `notebookNodeLogic` that is bound in the `NodeWrapper`

https://github.com/PostHog/posthog/assets/6685876/05336e46-5ec2-4330-a7d4-e645be5ac203

Still to do
- [ ] Create a notebook from a session recording
- [ ] Click the timestamp to scrub the video to the timestamp
- [ ] Show a suggestion for a timestamp when pressing Enter on a comment that can be `Tab`'ed for inclusion in the document

## How did you test this code?

- Manually (see gif)
- Would love to pair on adding some tests for notebooks if anyone has time

### CODE REVIEW

**Comment by daibhin** [comment]
Conscious that notebooks don't have a lot of tests in them. Would love to pair with someone if there is a way to start adding them in this PR

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeTimestamp.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import clsx from 'clsx'
+import { lastChildOfType } from '../Notebook/Editor'
+
+const Component = (props: NodeViewProps): JSX.Element => {
+    const playbackTime: number = props.node.attrs.playbackTime
+
+    return (
+        <NodeViewWrapper as="span" class={clsx('Timestamp', props.selected && 'Timestamp--selected')}>
+            {formatTimestamp(playbackTime)}
+        </NodeViewWrapper>
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeTimestamp.tsx:17]: Maybe more specific? `NotebookRecordingTimestamp` or something.

Timestamp feels to generic and could end up clashing with something else I would imagine...

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeTimestamp.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    )
+}
+
+function formatTimestamp(time: number): string {
+    return dayjs.duration(time, 'milliseconds').format('HH:mm:ss').replace(/^00:/, '').trim()
+}
+
+export const NotebookNodeTimestamp = Node.create({
+    name: NotebookNodeType.Timestamp,
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeTimestamp.tsx:26]: Same here - maybe `ReplayTimestamp` so its super clear what it is for?

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeTimestamp.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                const selectedNode = editor.state.selection.$head.parent
+                const timestampChild = lastChildOfType(selectedNode, NotebookNodeType.Timestamp)
+
+                if (selectedNode.type.name === 'paragraph' && timestampChild) {
+                    const sessionRecordingId = timestampChild.attrs.sessionRecordingId
+
+                    const currentPlayerTime =
+                        sessionRecordingPlayerLogic.findMounted(sessionRecordingPlayerProps(sessionRecordingId))?.values
+                            .currentPlayerTime || 0
+
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeTimestamp.tsx:58]: This is a great solution. 
We could after follow up with hover and click handlers to call the related logic and set some styles to indicate that it is related to the timestamp in question...

File: frontend/src/scenes/notebooks/Notebook/Editor.tsx [hunk 4]
```diff
@@ -72,12 +77,21 @@ export function Editor({
             NotebookNodePerson,
             NotebookNodeFlag,
             SlashCommandsExtension,
+            NotebookNodeTimestamp,
+            Extension.create({
+                addGlobalAttributes() {
+                    return [
+                        {
+                            types: [NotebookNodeType.Recording],
+                            attributes: { notebookShortId: { default: notebookId } },
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Notebook/Editor.tsx:86]: Hmmmm not sure that I love this... Could we not just use the notebookLogic instead? Every node is mounted within a notebookLogic so you should be able to easily grab the currently mounted notebookLogic and pull out the Id from that?

If I've missed something we can look at it together

**Review by benjackwhite — CHANGES_REQUESTED** [review]

**Comment by benjackwhite** [comment]
@daibhin resolved the comments that were sorted from our pairing. Just some small follow up stuff left 👍

**Review by benjackwhite — APPROVED** [review]
Very nice!

**Pull request merged**